game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1394,29 +1380,25 @@ private void removeNonCombatants(final IDelegateBridge bridge) {
     return unitList;
   }
 
-  private void landParatroops(final IDelegateBridge bridge) {
-    if (TechAttachment.isAirTransportable(attacker)) {
-      final Collection<Unit> airTransports =
-          CollectionUtils.getMatches(battleSite.getUnits(), Matches.unitIsAirTransport());
-      if (!airTransports.isEmpty()) {
-        final Collection<Unit> dependents = getDependentUnits(airTransports);
-        if (!dependents.isEmpty()) {
-          final CompositeChange change = new CompositeChange();
-          // remove dependency from paratroopers by unloading the air transports
-          for (final Unit unit : dependents) {
-            change.add(TransportTracker.unloadAirTransportChange(unit, battleSite, false));
-          }
-          bridge.addChange(change);
-          // remove bombers from dependentUnits
-          for (final Unit unit : airTransports) {
-            dependentUnits.remove(unit);
-          }
-        }
-      }
+  @Override
+  public void landParatroopers(
+      final IDelegateBridge bridge,
+      final Collection<Unit> airTransports,
+      final Collection<Unit> dependents) {
+    final CompositeChange change = new CompositeChange();
+    // remove dependency from paratroopers by unloading the air transports
+    for (final Unit unit : dependents) {
+      change.add(TransportTracker.unloadAirTransportChange(unit, battleSite, false));
+    }
+    bridge.addChange(change);
+    // remove bombers from dependentUnits
+    for (final Unit unit : airTransports) {
+      dependentUnits.remove(unit);
     }
   }
 
-  private void markNoMovementLeft(final IDelegateBridge bridge) {
+  @Override
+  public void markNoMovementLeft(final IDelegateBridge bridge) {
```

Comment:
Is there a plan to move this method to the step logic, or will it be here long-term?
It seems like if we are adding a step from `MustFightBattle` that then does all its work by invoking a method on `MustFightBattle` that we are not really benefiting a lot and have a lot of indirection. Hence I wonder if this is a temporary situation or perhaps there is a reason this can't be moved?

Reply:
Temporary situation.  See my comment above about my goals.